### PR TITLE
Add appveyor testing of Requests.

### PR DESCRIPTION
**Pull request by Lukasa**
In general it'd be good to have more testing, but it's also possible that via PR #3979 we'll get unexpected different behaviour on Windows than on other platforms. For this reason, we should start also testing on Windows.

This PR takes the configuration used for urllib3, updates it for Requests, and then provides it. We'll still need to actually enable Appveyor so that it'll try to build, which I cannot do as I don't have admin on the repo. @kennethreitz, mind giving that a try?

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# AppVeyor.yml from https://github.com/ogrisel/python-appveyor-demo
+# License: CC0 1.0 Universal: http://creativecommons.org/publicdomain/zero/1.0/
+
+build: off
+
+environment:
+  matrix:
+    - PYTHON: "C:\\Python266-x64"
```
Comment (sigmavirus24) [inline appveyor.yml:8]: Didn't we drop support for 2.6?

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    - PYTHON: "C:\\Python34-x64"
+      PYTHON_VERSION: "3.4.x"
+      PYTHON_ARCH: "64"
+      TOXENV: "py34"
+
+    - PYTHON: "C:\\Python35-x64"
+      PYTHON_VERSION: "3.5.x"
+      PYTHON_ARCH: "64"
+      TOXENV: "py35"
```
Comment (sigmavirus24) [inline appveyor.yml:31]: Is 3.6 not supported on AppVeyor?

File: appveyor.yml (hunk 1 above)
Comment (nateprewitt) [reply appveyor.yml:8]: Not until 3.0 was the decision I believe.

Comment (Lukasa) [reply appveyor.yml:8]: Correct.

File: appveyor.yml (hunk 2 above)
Comment (Lukasa) [reply appveyor.yml:31]: It should be, this was just an oversight.

Comment (Lukasa) [reply appveyor.yml:31]: Ok, added

**Comment by kennethreitz** [comment]
i'm very skeptical, but let's give it a shot

**Comment by kennethreitz** [comment]
doesn't travis have a windows option?

**Comment by kennethreitz** [comment]
build failed, looks like it's because of 2.6 maybe. https://ci.appveyor.com/project/kennethreitz/requests

**Comment by Lukasa** [comment]
Travis does not have a Windows option and does not include one in their roadmap.

**Comment by Lukasa** [comment]
So the reason those builds are failing is because I have to work out where the hell GNU make is on those systems. Given that our CI builds use make, I need to find where it is. If you're happy to be a bit patient, I'll open a PR to fix it in the next few hours

**Pull request merged**